### PR TITLE
Add support for Uniffi

### PR DESCRIPTION
**Pull request by FabioPinheiro**
[ATL-6717]
[ATL-6841]
[ATL-6842]
[ATL-6843]

[ATL-6717]: https://input-output.atlassian.net/browse/ATL-6717?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ATL-6841]: https://input-output.atlassian.net/browse/ATL-6841?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ATL-6842]: https://input-output.atlassian.net/browse/ATL-6842?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ATL-6843]: https://input-output.atlassian.net/browse/ATL-6843?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: output-frameworks/demo-scala/native-lib/NATIVE/linux-x86-64/libuniffi_sd_jwt.so
Comment (hamada147) [inline output-frameworks/demo-scala/native-lib/NATIVE/linux-x86-64/libuniffi_sd_jwt.so:1]: Please don't merge while the compiled version exist

File: src/disclosure.rs [hunk 1]
```diff
@@ -6,20 +6,20 @@ use crate::utils::generate_salt_mock;
 use serde_json::Value;
 
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```
Comment (goncalo-frade-iohk) [inline src/disclosure.rs]: Mate unfortunately this is something we cannot do. For maintainability issues.

We want to be able to merge with updates they do, so we cannot change the source code. Rather we need to make wrappers around it as needed, on our Uniffi project.

PS: The comment is here but it applies to any change in this folder.

**Review by goncalo-frade-iohk — CHANGES_REQUESTED** [review]

**Comment by mkbreuningIOHK** [comment]
@hamada147 , can you add the reference ATL-6717 in the PR comment?

File: output-frameworks/demo-scala/native-lib/NATIVE/linux-x86-64/libuniffi_sd_jwt.so
Comment (FabioPinheiro) [inline output-frameworks/demo-scala/native-lib/NATIVE/linux-x86-64/libuniffi_sd_jwt.so:1]: Let's not commit the binaries

File: output-frameworks/demo-scala/sd-jwt-jvm-1.0-SNAPSHOT.jar
Comment (FabioPinheiro) [inline output-frameworks/demo-scala/sd-jwt-jvm-1.0-SNAPSHOT.jar:1]: Same here.Let's not commit the binaries

Comment (hamada147) [reply output-frameworks/demo-scala/sd-jwt-jvm-1.0-SNAPSHOT.jar:1]: Commit by mistake, gonna do a rebase and delete them

File: output-frameworks/demo-scala/native-lib/NATIVE/linux-x86-64/libuniffi_sd_jwt.so
Comment (hamada147) [reply output-frameworks/demo-scala/native-lib/NATIVE/linux-x86-64/libuniffi_sd_jwt.so:1]: Commit by mistake, gonna do a rebase and delete them

**Comment by Dale-Doe** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [cdf61bbd\-9355\-4b5d\-acb6\-a841c73d326c](https://eu.ast.checkmarx.net/projects/5d4491d1-a62b-45db-b93c-b978026ab8e7/scans?id=cdf61bbd-9355-4b5d-acb6-a841c73d326c&branch=uniffi)
### New Issues
Severity|Issue|Source File / Package|Checkmarx Insight
---|---|---|---
LOW|[Missing_Rooted_Device_Check](https://eu.ast.checkmarx.net/results/5d4491d1-a62b-45db-b93c-b978026ab8e7/cdf61bbd-9355-4b5d-acb6-a841c73d326c/sast/description/693/5904646516379822605)|/sd\-jwt\-kmp/build\.gradle\.kts: [57](https://github.com/input-output-hk/sd-jwt-rust/blob/uniffi//sd-jwt-kmp/build.gradle.kts#L57)|[Attack Vector](https://eu.ast.checkmarx.net/results/cdf61bbd-9355-4b5d-acb6-a841c73d326c/5d4491d1-a62b-45db-b93c-b978026ab8e7/sast?result-id=27m3VMZZ4U4Rq3nxliIcT0jqmkg%3D&redirect=true)

**Comment by FabioPinheiro** [comment]
The force push to the branch is a bit inconvenient

File: Cargo.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "sd-jwt-rs"
+name = "sd_jwt_rs"
```
Comment (FabioPinheiro) [inline Cargo.toml]: In Java this import is not convenient.
`import sd_jwt_rs.*`

Let`s pick another name without camel case or underscores

**Comment by FabioPinheiro** [comment]
We should add to the README the command you are using to build, +- as I had before.

**Review by FabioPinheiro — COMMENTED** [review]
I want
 to test the Jar but LGTM

**Review by goncalo-frade-iohk — APPROVED** [review]

**Review by mineme0110 — APPROVED** [review]

**Pull request merged**